### DISCUSSION TITLE
Plan to release source for Codex

### DISCUSSION THREAD

**Discussion by DelphiWorlds** [DelphiWorlds/Codex#27] [Announcements]
I have a plan to release the source code for Codex by the time v2.0.0 is due for release, which is planned for late this year (2023)

This will happen only once I've resolved a few dependencies, which includes JCL/JVCL:
 
Codex uses TJvCreateProcess in order to execute commands such as some of the Android tools, e.g. aapt/aapt2, javac etc. I'd like to completely remove this dependency, and have alternative [code which is Work In Progress in the Playground](https://github.com/DelphiWorlds/Playground/tree/main/Demos/RunProcess), however as per the comments in the code and the [reported issue](https://github.com/DelphiWorlds/Playground/issues/17), I am having trouble making it work. 

If you are able to help resolve this issue, it will make the release of the source for Codex a reality.

Thanks in advance for any help.

**Comment by War3Evo** [comment]
Can you supply a early release of source for Java2OP and Build Jar?

**Reply by DelphiWorlds** [reply]
Java2OP itself is included with Delphi (in the `bin\converters\java2op` folder). Codex just has a convenient front-end for this command where you can easily select which classes to import, from which packages. It can also do some post-processing that cleans up the import. 

I'll take a look to see if I can make this part of the code (as well as for the Build Jar function) available separate to the Codex repo

**Reply by DelphiWorlds** [reply]
This is a gist for the part of Codex that does the actual work of compiling Java source into a jar:
https://gist.github.com/DelphiWorlds/9043739d2e4629dfa3df8a17723a03a3
This is a gist for the part of Codex that executes Java2OP and performs post-processing:
https://gist.github.com/DelphiWorlds/481e54060ce9c18f253f37fd799df1fc

**Comment by War3Evo** [comment]
The reported issue for what Antivirus?

**Reply by DelphiWorlds** [reply]
AV in this case means Access Violation. I've since resolved that issue

**Comment by War3Evo** [comment]
If you just need to execute command line stuff, I've got a threaded component that will do that.

It will execute command line stuff, and you don't have to show the information anywhere.  It's optional to show any information or outputs.

I've used this to do windows ssh stuff.

https://github.com/War3Evo/gtDelphi/blob/master/CommandPrompt/gtCommandPrompt.pas

**Reply by DelphiWorlds** [reply]
I already had working code that executes command line executables, using TJvCreateProcess from JVCL. I wanted to remove the dependency on JVCL, so started working on alternative code. I now have [code that works](https://github.com/DelphiWorlds/Playground/tree/main/Demos/RunProcess), however thanks for the link to yours.
